devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -138,47 +151,80 @@ private parseAttrMessage(description) {
 	}
 	attrData.each {
 		def map = [:]
-		if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == LOCAL_TEMPERATURE) {
-			log.debug "TEMP"
-			map.name = "temperature"
-			map.value = getTemperature(it.value)
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == COOLING_SETPOINT) {
-			log.debug "COOLING SETPOINT"
-			map.name = "coolingSetpoint"
-			map.value = getTemperature(it.value)
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == HEATING_SETPOINT) {
-			log.debug "HEATING SETPOINT"
-			map.name = "heatingSetpoint"
-			map.value = getTemperature(it.value)
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE)) {
-			log.debug "MODE"
-			map.name = "thermostatMode"
-			map.value = THERMOSTAT_MODE_MAP[it.value]
-			map.data = [supportedThermostatModes: state.supportedModes]
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == THERMOSTAT_RUNNING_STATE) {
-			log.debug "RUNNING STATE"
-			def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
-			map.name = "thermostatOperatingState"
-			if(binValue[0] == "1") {
-				map.value = "heating"
-			} else if(binValue[1] == "1") {
-				map.value = "cooling"
-			} else {
-				map.value = binValue[2] == "1" ? "fan only" : "idle"
+		if (it.cluster == THERMOSTAT_CLUSTER) {
+			if (it.attribute == LOCAL_TEMPERATURE) {
+				log.debug "TEMP"
+				map.name = "temperature"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == COOLING_SETPOINT) {
+				log.debug "COOLING SETPOINT"
+				map.name = "coolingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == HEATING_SETPOINT) {
+				log.debug "HEATING SETPOINT"
+				map.name = "heatingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE) {
+				log.debug "MODE"
+				map.name = "thermostatMode"
+				map.value = THERMOSTAT_MODE_MAP[it.value]
+				map.data = [supportedThermostatModes: state.supportedThermostatModes]
+			} else if (it.attribute == THERMOSTAT_RUNNING_STATE) {
+				log.debug "RUNNING STATE"
+				def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
+				map.name = "thermostatOperatingState"
+				if (binValue[0] == "1") {
+					map.value = "heating"
+				} else if (binValue[1] == "1") {
+					map.value = "cooling"
+				} else {
+					map.value = binValue[2] == "1" ? "fan only" : "idle"
+				}
+			} else if (it.attribute == THERMOSTAT_SYSTEM_CONFIG) {
+				def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
```

Comment:
I think it would be more clear to convert to a number instead of leaving it as a string

Reply:
yeah i think bitmasking is the better way to go here too

Reply:
Finishing that up; same for THERMOSTAT_RUNNING_STATE

Reply:
Done!